### PR TITLE
drivers: usb_dc_nrfx: move to new DT API

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Close out the existing partial conversion to the new DT API.

Tested on nRF52840-DK.

### CODE REVIEW

File: drivers/usb/device/usb_dc_nrfx.c [hunk 1]
```diff
@@ -29,6 +29,10 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(usb_nrfx);
 
+/* USB device controller access from devicetree */
+#define USBD_NODE DT_INST(0, nordic_nrf_usbd)
```
Comment (jfischer-phytec-iot) [inline drivers/usb/device/usb_dc_nrfx.c:33]: btw, I can not remember exactly but I think the consensus was to use just usb without d, https://github.com/zephyrproject-rtos/zephyr/pull/24362/commits/d7787dff535c99e654f7afa229cb40190af21191#diff-2b6788bec1db035ecf36cf22c40c6a70R46

Comment (galak) [reply drivers/usb/device/usb_dc_nrfx.c:33]: The one's you reference are changed to match the SoC's manual.

**Review by galak — CHANGES_REQUESTED** [review]
If you are going to use INST variants we should use DT_INST macro's.

**Comment by mbolivar-nordic** [comment]
> If you are going to use INST variants we should use DT_INST macro's.

fair enough

**Review by anangl — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@galak this one should be ready now. Usual unrelated mps2_an385 failure.

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
CI failure due to qemu mps2 which isn't related.

**Pull request merged**